tests/frontend/test_engine.py

```diff
@@ -164,228 +144,150 @@ def test_print_commands(self, backend):
 class TestRegRefs:
     """Testing register references."""
 
-    def test_wrong_index(self):
+    def test_corrupt_index(self, prog):
         """This should never happen!"""
-        eng, q = sf.Engine(2)
-        q[0].ind = 1
-
-        with eng:
-            with pytest.raises(engine.RegRefError, match="Should never happen!"):
+        with prog.context as q:
+            q[0].ind = 1
+            with pytest.raises(program.RegRefError, match="Should never happen!"):
```

Comment:
Will need to update once we update the error message to be more explicit about the behaviour that should never happen